nodes/nomos-node-api/src/lib.rs

```diff
@@ -62,9 +62,14 @@ where
     }
 
     /// Service main loop
-    async fn run(mut self) -> Result<(), DynError> {
-        let endpoint = B::new(self.settings.backend_settings).await?;
-        endpoint.serve(self.handle).await?;
+    async fn run(mut self) -> Result<(), ServiceError> {
+        let endpoint = B::new(self.settings.backend_settings)
+            .await
+            .map_err(|e| Box::new(e) as DynError)?;
```

Comment:
Maybe we could make a method? so we do `map_err(ServiceError::from_whatever)`? Which can be generic?

Reply:
Yeah, I change it back to `ServiceError::service`.